### ISSUE TITLE
task.unique with kill_me sometimes kills itself when it shouldn't

### ISSUE THREAD

**Issue by dlashua** [custom-components/pyscript#40]
code:
```python
import time

rt = []

class TaskUniqueTest:

    def kill_others(self):
        log.info(f'{self.name}: killing others')
        task.unique('task_unique_test')
        log.info(f'{self.name}: survived')

    def kill_myself(self):
        log.info(f'{self.name}: killing myself')
        task.unique('task_unique_test', kill_me=True)
        log.info(f'{self.name}: survived (there were no others)')

    def __init__(self, name, seconds):
        self.name = name
        self.seconds = seconds
        log.info(f'{self.name}: starting {__name__}')
        self.kill_others()

        @time_trigger('startup')
        def inner_startup():
            log.info(f'{self.name}: in inner_startup')
            log.info(f'{self.name}: sleeping {self.seconds}')
            start_time = time.time()
            task.sleep(self.seconds)
            elapsed_time = round((time.time() - start_time), 2)
            log.info(f'{self.name}: done sleeping {elapsed_time}/{self.seconds}')
            log.info(f'{self.name}: calling update')
            self.update()

        rt.append(inner_startup)

    def update(self):
        log.info(f'{self.name}: in update')
        self.kill_myself()
        log.info(f'{self.name}: i did not die')



@time_trigger('startup')
def startup():
    task.sleep(2)
    a = TaskUniqueTest('a', 10)
    task.sleep(2)
    b = TaskUniqueTest('b', 5)

```

logs:
```
2020-10-12 12:53:04 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] a: starting apps.for_test
2020-10-12 12:53:04 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] a: killing others
2020-10-12 12:53:04 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] a: survived
2020-10-12 12:53:04 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: in inner_startup
2020-10-12 12:53:04 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: sleeping 10
2020-10-12 12:53:06 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] b: starting apps.for_test
2020-10-12 12:53:06 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] b: killing others
2020-10-12 12:53:06 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] b: survived
2020-10-12 12:53:06 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: in inner_startup
2020-10-12 12:53:06 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: sleeping 5
2020-10-12 12:53:11 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: done sleeping 5.0/5
2020-10-12 12:53:11 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: calling update
2020-10-12 12:53:11 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: in update
2020-10-12 12:53:11 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: killing myself
2020-10-12 12:53:11 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: survived (there were no others)
2020-10-12 12:53:11 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: i did not die
2020-10-12 12:53:14 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: done sleeping 10.0/10
2020-10-12 12:53:14 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: calling update
2020-10-12 12:53:14 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: in update
2020-10-12 12:53:14 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: killing myself
2020-10-12 12:53:14 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: survived (there were no others)
2020-10-12 12:53:14 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: i did not die
```

Since "b" killed others with `task.unique()` while "a" was sleeping, "a" should have been killed. However, "a" completes. Additionally, if "a" was still a running task, then when "b" killed itself with `task.unique(kill_me=True)`, it should not have survived.

The expected outcome of the above is:
a starts
a calls task.unique which kills nothing
a sleeps
b starts
b calls task.unique which kills a
b sleeps
b completes

**Comment by dlashua** [comment]
actually, this is okay. Since "inner_startup" is a different task that never calls task.unique, it's not killed.

However, when I move kill_others() into inner_startup, this happens:

```
2020-10-12 13:13:10 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] a: starting apps.for_test
2020-10-12 13:13:10 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: killing others
2020-10-12 13:13:10 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: survived
2020-10-12 13:13:10 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: in inner_startup
2020-10-12 13:13:10 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] a: sleeping 10
2020-10-12 13:13:12 INFO (MainThread) [custom_components.pyscript.apps.for_test.startup] b: starting apps.for_test
2020-10-12 13:13:12 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: killing others
2020-10-12 13:13:12 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: survived
2020-10-12 13:13:12 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: in inner_startup
2020-10-12 13:13:12 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: sleeping 5
2020-10-12 13:13:17 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: done sleeping 5.0/5
2020-10-12 13:13:17 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: calling update
2020-10-12 13:13:17 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: in update
2020-10-12 13:13:17 INFO (MainThread) [custom_components.pyscript.apps.for_test.inner_startup] b: killing myself
```

"b" would have been the only task running, so when it killed itself, it should have survived. But it didn't.

I believe the reason is that that one task called both `task.unique()` and `task.unique(kill_me=True)`.

Line 111 of `functions.py` should do:
```python
if kill_me and cls.unique_name2task[name] != asyncio.current_task():
```

Or, something to that effect.

**Comment by craigbarratt** [comment]
Good catch.  Thanks for the self-contained examples.  Should be fixed now with 3426454.

**Comment by dlashua** [comment]
I tested this several ways and it seems to be working correctly. Thank you!!
